### ISSUE TITLE
Can't specify literal values for | or : 

### ISSUE THREAD

**Issue by Truncated** [mugiwara85/CodeblockCustomizer#93]
The parsing of the `abc:xyz` and | for constraining lines is nice. However I'm trying to highlight starting or ending on colons and pipes for regex and it doesn't appear that I can start or end on spaces, pipes, or colons as literal values.

Here's an example of one of the documentation lines I am trying to highlight to demonstrate how the reg-ex groups apply:
`{(workflow)(:name=.*??)(\\|.*??})`

`{workflow:name=any combo of characters until the first pipe...|after which any combo of characters until an end bracket...}`

To highlight three parts of this:
`workflow`
`:name=any combo of characters until the first pipe...`
`|after which any combo of characters until an end bracket...}`

Obviously, straight forward as-is you might want:

hl:workflow,::...,|:}

...of course, that didn't work at all... I tried a bunch of combinations which came up with some interesting results:
![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/14208325/3e003a43-0dbf-44ec-adeb-a1f4106c46f9)
![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/14208325/255af9e2-bb52-47cd-b0c6-35b5bada4332)
![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/14208325/b69b1aa3-6fbd-4c60-af30-6a408aa80d4b)
![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/14208325/47438caf-029c-4cc0-a7e9-5cd642ca114d)
![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/14208325/6836ea7e-887d-4d05-a744-aaa9319495bf)
![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/14208325/a1e901f4-6f54-40c1-9c8c-24245e502e48)

Either being able to optionally wrap `abc:xyz` in quotes to be explicit a'la `"abc":"xyz"` or maybe implementing some way to escape the characters for literal use would be helpful. 

With quotes, it might become:
`hl:"workflow",":":"...","|":"}"`

As an imperfect workaround, when using the alternative colors (especially if they aren't transparent), the layering did allow me to fake it with:
G1:workflow G2:w:... G5:.|:} 
![image](https://github.com/mugiwara85/CodeblockCustomizer/assets/14208325/c2b191e1-e9c9-4b10-9bdb-5418a64f5169)

**Comment by Truncated** [comment]
Couple other characters can't be used to start the abc:xyz elements include:

dash (-) and double-quotes (")

**Comment by mugiwara85** [comment]
yeah, this might have to do how I parse the string. I will look at it, and get back to you.

**Comment by Truncated** [comment]
I think it depends on whether your library / API has a command for "automatically escape anything that needs it to take this string literally as-is" without having to make that parsing and swapping yourself. :)

**Comment by mugiwara85** [comment]
@Truncated This was also implemented. Unfortunately, this was more problematic as it seems, but I solved using a workaround. 
The reason you couldn't specify `|` or `:` for highlighting is, because these chars are used to split the parameter, to get the values. There could be a lot of problems with, for example if you want to do something like: `hlt:string:sdf:sdf:fg` Which `:` would here mean the separator? And many other cases.

But to cut to the chase, I did the following. In the settings page there is a line sperator, and text separator option, where you can define them globally. You can also use the `lsep` (Line Separator) and `tsep` (Text separator) option to define these specifically for code blocks. So for example if you want to highlight text which has a `:` in it, you could do something like : `hlt:mystring:/end lsep:/` This way the string would be splitted at the `/` char and not at `:`.

Hopefully, this is ok, but I think this way everyone can cover every possible case.

As always, I would appreciate a good test, and a feedback before releasing it.

**Comment by Truncated** [comment]
Will do!

**Comment by mugiwara85** [comment]
Oh wait, I found a few bugs. Don't test it yet please.

**Comment by mugiwara85** [comment]
@Truncated sorry for the delay, but now you can test it! Editing view and reading mode are fully functional

**Comment by mugiwara85** [comment]
Should be fixed now in version 1.2.8
